lib/api/2.0/schemas2.js

```diff
@@ -26,26 +28,25 @@ var schemasIdGet = controller(function(req) {
 
 // GET /api/2.0/schemas/tasks
 var taskSchemasGet = controller(function () {
-    return taskOptionValidator.getAllSchemaNames({ includeNameSpace: false });
+    return workflowApiService.getTaskDefinitions()
+    .then(function(tasks) {
```

Comment:
The promise that returned from `getTaskDefinition` is not bluebird style promise, it is waterline's promise, it doesn't have  a `map` function, so that's the reason I ended up using the array's `map`.

Now in RackHD code there is various Promise (at least there are Bluebird promise, node.js native promise, waterline promise), we'd better to combine them to a single one in future.